ecl/hthor/hthor.cpp

```diff
@@ -4929,7 +4953,7 @@ bool CHThorSelfJoinActivity::fillGroup()
     failingOuterAtmost = false;
     OwnedConstRoxieRow next;
     unsigned groupCount = 0;
-    next.setown(input->nextInGroup());
+    next.setown(groupedInput->nextInGroup());
```

Comment:
I don't think this is correct for the limited prefixed join case - previously it would have read from dualCacheInput